quill-core/src/test/scala/io/getquill/sources/SourceMacroSpec.scala

```diff
@@ -124,4 +124,14 @@ class SourceMacroSpec extends Spec {
     var db = source(new MirrorSourceConfig(""))
     "db.run(qr1)" must compile
   }
+  
+  "fails if bind value is outside `source.run`" in {
+    val q = {
+      val i = 1
```

Comment:
> I don't see a passing test for motivation of this PR.

I've added tests with inline queries.

> In other words, what makes val i = 1 a free variable and p1 in helper method not?

`val i = 1` isn't in the scope of `db.run`, that's why it can't be used as a free variable. Free variables can't be used for query composition, only within `db.run`.